lib/filterx/func-flags.h

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2023 Axoflow
+ * Copyright (c) 2024 shifter
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#ifndef FUNC_FLAGS_H
+#define FUNC_FLAGS_H
+
+#define MAX_ENUMS 64
+
+#define STATIC_ASSERT(COND, MSG) _Static_assert(COND, MSG)
```

Comment:
Let's use `G_STATIC_ASSERT` instead, it's more compatible with older C versions.
(`syslog-ng.h` should be included instead of glib directly due to some glib compat headers we have)